### PR TITLE
feat: adds utility service and access to commander

### PR DESCRIPTION
**Pull request by jmcdo29**
There is a new `CliUtilityService` and `@InjectCommnder()` decoraotr
    that allows for direct access to the commander instnace. The utility service
    has methods like `parseBoolean`, `parseInt`, and `parseFloat`. The
    number parsing methods are just simple wrappers around `Number.parse*()`,
    but the boolean parsing method handles true being `yes`, `y`, `1`,
    `true`, and `t` and false being `no`, `n`, `false`, `f`, and `0`.

### CODE REVIEW

**Comment by jmcdo29** [comment]
@micalevisk @smolinari would y'all mind taking a look at this and checking my logic along with what the addition is? Thanks guys

File: apps/docs/docs/api.md [hunk 1]
```diff
@@ -154,6 +154,39 @@ Most of the decorators for the `@QuestionSet()` decorated class are just dynamic
...
+
+:::
+
+#### parseBoolean
+
+A helper method to turn a string into a boolean based on true or false values.
+
+| Property | Type | Required | Description |
+| --- | --- | --- | --- |
+| val | string | true | The string value to be turned into a boolean |
```
Comment (micalevisk) [inline apps/docs/docs/api.md:171]:
```suggestion
| val | `string` | true | The string value to be turned into a boolean |
```

and so on

**Comment by micalevisk** [comment]
I'll try this feat out next saturday. The logic seems fine.

---

I think would be cool to add in-code docs like

```js
/**
 * A helper method to turn a string (case insensitive) into a boolean based on true or false values.
 *
 * Any string that matches one of the following values is considered a `true` input:
 * - `yes`
 * - `y`
 * - `true`
 * - `t`
 * - `1`
 *
 * Any string that matches one of the following values is considered a `false` input:
 * - `no`
 * - `n`
 * - `false`
 * - `f`
 * - `0`
 *
 * @param val
 * @throws {Error} If `val` is not a _true_ nor _false_ expression
 */
parseBoolean(val: string): boolean {
```

and

```js
/**
 * A Simple wrapper around `Number.parseInt()`
 * @param val
 * @param radix
 */
parseInt(val: string, radix = 10): number {

/**
 * A simple wrapper around `Number.parseFloat()`
 * @param val
 */
parseFloat(val: string): number {
```

to improve DX a bit. Although this is redundant as they're documented already on the docs site.

**Comment by jmcdo29** [comment]
> I'll try this feat out next saturday. The logic seems fine.
> 
> I think would be cool to add in-code docs like
> 
> ```js
> /**
>  * A helper method to turn a string (case insensitive) into a boolean based on true or false values.
>  *
>  * Any string that matches one of the following values is considered a `true` input:
>  * - `yes`
>  * - `y`
>  * - `true`
>  * - `t`
>  * - `1`
>  *
>  * Any string that matches one of the following values is considered a `false` input:
>  * - `no`
>  * - `n`
>  * - `false`
>  * - `f`
>  * - `0`
>  *
>  * @param val
>  * @throws {Error} If `val` is not a _true_ nor _false_ expression
>  */
> parseBoolean(val: string): boolean {
> ```
> 
> and
> 
> ```js
> /**
>  * A Simple wrapper around `Number.parseInt()`
>  * @param val
>  * @param radix
>  */
> parseInt(val: string, radix = 10): number {
> 
> /**
>  * A simple wrapper around `Number.parseFloat()`
>  * @param val
>  */
> parseFloat(val: string): number {
> ```
> 
> to improve DX a bit. Although this is redundant as they're documented already on the docs site.

Yeah, I think the doc string would definitely be helpful for in IDE docs. It's just that none of the other methods have that, so adding it might feel out of place.

**Comment by smolinari** [comment]
From what I can see, looks good. 👍 Asking me to check your work is like master Po asking Grasshopper if his Kung Fu is good. LOL! 😁 

Scott

**Comment by micalevisk** [comment]
@jmcdo29 how do we import this new service?

it isn't listed below

image

**Comment by jmcdo29** [comment]
Whoops, missed that. See this is why I have someone double check me

**Review by micalevisk — APPROVED** [review]

**Comment by coler-j** [comment]
How do we use this?

**Comment by jmcdo29** [comment]
@coler-j [Is there something not understood about the documentation](https://nest-commander.jaymcdoniel.dev/en/features/utility/)?

**Pull request closed without merging**